src/persist-client/src/bin/persist_open_loop_benchmark.rs

```diff
@@ -0,0 +1,449 @@
+                        return Ok(finished);
+                    }
+                };
+
+                trace!("writer {} received a batch. writing", idx);
+                let write_start = Instant::now();
+                writer.write(batch).await?;
+                batches_written += 1;
+                let write_latency = write_start.elapsed();
+                if write_latency > max_write_latency {
+                    max_write_latency = write_latency;
+                }
+
+                let elapsed = start.elapsed();
+
+                if elapsed - prev_log > FIVE_SECONDS {
+                    let records_written = batches_written * args.batch_size;
+                    info!("After {} ms writer {} has written {} records. Max write latency {} ms most recent write latency {} ms.",
+                          elapsed.as_millis(), idx, records_written, max_write_latency.as_millis(), write_latency.as_millis());
+                    prev_log = elapsed;
+                }
+            }
+        });
+
+        handles.push(writer_handle);
+    }
+
+    for (idx, mut reader) in readers.into_iter().enumerate() {
+        let b = Arc::clone(&barrier);
+        let reader_handle = mz_ore::task::spawn(|| format!("reader-{}", idx), async move {
+            trace!("reader {} waiting for barrier", idx);
+            b.wait().await;
+            info!("starting reader {}", idx);
+
+            // Max observed latency for BenchmarkReader::num_records.
+            let mut max_read_latency = Duration::from_millis(0);
+
+            // Max observed delay between the number of records expected to be read at any
+            // point in time vs the number of records actually ingested by that point.
+            let mut max_lag = 0;
+
+            // The last time we emitted progress information to stdout, expressed
+            // as a relative duration from start.
+            let mut prev_log = Duration::from_millis(0);
+            loop {
+                let elapsed = start.elapsed();
+                let expected_sent =
+                    elapsed.as_millis() as usize / (millis_per_batch as usize) * args.batch_size;
+                let read_start = Instant::now();
+                let num_records_read = reader.num_records().await?;
+                let read_latency = read_start.elapsed();
+                let lag = if expected_sent > num_records_read {
+                    expected_sent - num_records_read
+                } else {
+                    0
+                };
+                if lag > max_lag {
+                    max_lag = lag;
+                }
+
+                if read_latency > max_read_latency {
+                    max_read_latency = read_latency;
+                }
+
+                if elapsed - prev_log > FIVE_SECONDS {
+                    info!("After {} ms reader {} has read {} records. Max read lag {} records, most recent read lag {} records. Max read latency {} ms, most recent read latency {} ms",
```

Comment:
let compute overall goodput since the start of the run and log it here as well. as I mentioned offline, i'm somewhat skeptical of read latency's usefulness here. so I think it's fine to include it, if this starts getting to be too much info, that's the thing I'd cut first

Reply:
done.